components/gitpod-cli/cmd/credential-helper.go

```diff
@@ -203,21 +199,23 @@ func walkProcessTree(pid int, fn pidCallbackFn) error {
 	}
 }
 
+var procStatPidRegExp = regexp.MustCompile(`\d+\ \(.+?\)\ .+?\ (\d+)`)
+
 func getProcesParentId(pid int) (ppid int, err error) {
 	statsString, err := readProc(pid, "stat")
 	if err != nil {
 		return
 	}
 
-	stats := strings.Fields(statsString)
-	if len(stats) < 3 {
+	match := procStatPidRegExp.FindStringSubmatch(statsString)
```

Comment:
how would this code need to look like so that we could introduce unit tests?

Reply:
I'd prefer to use this library https://github.com/prometheus/procfs for reading proc files, would that be Ok?

Reply:
Cool, I'll create another PR later cleaning this up